edb/lib/cal.edgeql

```diff
@@ -929,19 +1095,44 @@ CREATE INFIX OPERATOR
 std::`-` (l: cal::relative_duration, r: cal::relative_duration) -> cal::relative_duration {
     CREATE ANNOTATION std::identifier := 'minus';
     CREATE ANNOTATION std::description :=
-        'Time interval and date/time subtraction.';
+        'Time interval subtraction.';
     SET volatility := 'Immutable';
     USING SQL $$
     SELECT ("l"::interval - "r"::interval)::edgedb.relative_duration_t;
     $$;
 };
 
 
+CREATE INFIX OPERATOR
+std::`+` (l: cal::date_duration, r: cal::date_duration) -> cal::date_duration {
+    CREATE ANNOTATION std::identifier := 'plus';
+    CREATE ANNOTATION std::description :=
+        'Time interval addition.';
+    SET volatility := 'Immutable';
+    SET commutator := 'std::+';
+    USING SQL $$
+    SELECT ("l" + "r")::edgedb.date_duration_t;
+    $$;
+};
+
+
+CREATE INFIX OPERATOR
+std::`-` (l: cal::date_duration, r: cal::date_duration) -> cal::date_duration {
+    CREATE ANNOTATION std::identifier := 'minus';
+    CREATE ANNOTATION std::description :=
+        'Time interval subtraction.';
+    SET volatility := 'Immutable';
+    USING SQL $$
+    SELECT ("l" - "r")::edgedb.date_duration_t;
+    $$;
+};
```

Comment:
I don't see <, >, <=, >= defined for date_duration... are they defined automatically?

Reply:
The implicit cast from `date_duration` to `relative_duration` makes comparison operators compatible with both types (or even a mix, which is desirable). Since they return `bool` and operate on `interval` under the hood, having specialized versions doesn't really provide any special benefit.